UI.py

```diff
@@ -355,9 +355,8 @@ def CreateGetEndOfMonthReportEmbed(members_list, month = None, year = None):
 
 def FormatGetEndOfMonthReportEmbed(member, month, year):
     member_data = ""
-    previous_month = 12 if month == 1 else month - 1
-    start_date = datetime.datetime(year, previous_month, int(os.getenv("End_of_Month_Report_Day")))
-    end_date = datetime.datetime(year, month, int(os.getenv("End_of_Month_Report_Day")) - 1)
+    start_date = datetime.datetime(year, month, int(os.getenv("End_of_Month_Report_Day"))) - datetime.timedelta(days = utils.GetMonthDaysCount(month, year))
```

Comment:
doesn't do quite what you want it to i think. you want the monthdayscount of the previous month, not the current

Reply:
Done